cluster/gce/windows/k8s-node-setup.psm1

```diff
@@ -922,24 +907,60 @@ function Configure-GcePdTools {
     New-Item -path $PsHome\profile.ps1 -type file
   }
 
-    Add-Content $PsHome\profile.ps1 `
+  Add-Content $PsHome\profile.ps1 `
 '$modulePath = "K8S_DIR\GetGcePdName.dll"
 Unblock-File $modulePath
 Import-Module -Name $modulePath'.replace('K8S_DIR', ${env:K8S_DIR})
 }
 
-# Setup cni network. This function supports both Docker
-# and containerd.
-function Configure-CniNetworking {
+# Setup cni network. This function supports both Docker and containerd.
+function Prepare-CniNetworking {
   if (${env:CONTAINER_RUNTIME} -eq "containerd") {
+    # For containerd the CNI binaries have already been installed along with
+    # the runtime.
     Configure_Containerd_CniNetworking
   } else {
+    Install_Cni_Binaries
     Configure_Dockerd_CniNetworking
   }
 }
 
-# Downloads the Windows CNI binaries and writes a CNI config file under
-# $env:CNI_CONFIG_DIR.
+# Downloads the Windows CNI binaries and puts them in $env:CNI_DIR.
+function Install_Cni_Binaries {
+  if (-not (ShouldWrite-File ${env:CNI_DIR}\win-bridge.exe) -and
+      -not (ShouldWrite-File ${env:CNI_DIR}\host-local.exe)) {
+    return
+  }
+
+  $tmp_dir = 'C:\cni_tmp'
+  New-Item $tmp_dir -ItemType 'directory' -Force | Out-Null
+
+  $release_url = ('https://storage.googleapis.com/k8s-artifacts-cni/' +
+      'release%2f' + $CNI_RELEASE_VERSION + '%2f')
```

Comment:
i think %2f can be replaced by '/'

Reply:
also this is changed to env variable, it needs to be merge/resolve

Reply:
Rebased + resolved.